jme3-glsl-highlighter/src/com/jme3/gde/glsl/highlighter/lexer/GlslLexer.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package com.jme3.gde.glsl.highlighter.lexer;
+
+import java.util.logging.Logger;
+import org.netbeans.api.lexer.Token;
+import org.netbeans.spi.lexer.Lexer;
+import org.netbeans.spi.lexer.LexerInput;
+import org.netbeans.spi.lexer.LexerRestartInfo;
+import org.netbeans.spi.lexer.TokenFactory;
+
+/**
+ * Tokenizes the text the user is typing, so we can highlight it as needed.
+ * @author grizeldi
+ */
+public class GlslLexer implements Lexer<GlslTokenID>{
+    private final LexerInput lexerInput;
+    private final TokenFactory tokenFactory;
+    private final Logger log = Logger.getLogger(this.getClass().getCanonicalName());
+    
+    private String thisLineSoFar = "";
+
+    public GlslLexer(LexerRestartInfo info) {
+        lexerInput = info.input();
+        tokenFactory = info.tokenFactory();
+    }
+
+    @Override
+    public Token<GlslTokenID> nextToken() {
+        int c;
+        c = lexerInput.read();
+        thisLineSoFar += (char)c;
+        if (isDigit(c)){
+            while (true){
```

Comment:
Instead of endless loops, maybe add something like `while (c != LexerInput.EOF)`.
E.g. what happens if you enter a number with a sudden EOF before the Semicolon, endless loop?
But I guess then isDigit(next) evaluates to false in L39.

Just a thing to keep in mind though, in case hang ups occur